htlcswitch/link.go

```diff
@@ -714,6 +720,23 @@ func (l *channelLink) syncChanStates() error {
 			fundingLockedMsg := lnwire.NewFundingLocked(
 				l.ChanID(), nextRevocation,
 			)
+
+			// For option-scid-alias or zero-conf channels, ensure
+			// that we send over the alias in the funding_locked
+			// message.
+			scid := l.ShortChanID()
+			otherScid := l.OtherShortChanID()
+			if IsAlias(scid) {
+				// If the ShortChannelID is an alias, then
+				// this is a zero-conf channel.
+				fundingLockedMsg.AliasScid = &scid
+			} else if otherScid != hop.Source {
+				// If the ShortChannelID is not an alias, and
+				// the OtherShortChanID is not the default,
+				// then this is an option-scid-alias channel.
+				fundingLockedMsg.AliasScid = &otherScid
```

Comment:
Still getting through the PR, but leaving a not here that we should either have an option for auto rotation, or also let users modify the scid mapping over RPC so they can implement their own rotation.